src/init.cpp

```diff
@@ -704,14 +704,14 @@ void ThreadImport(std::vector<boost::filesystem::path> vImportFiles)
             LogPrintf("Headers-only reindexing finished. Going on with blocks\n");
 
         nFile = 0;
-        uiInterface.InitMessage(_("Reindexing block from files..."));
         while (true)
         {
             CDiskBlockPos pos(nFile, 0);
             if (!boost::filesystem::exists(GetBlockPosFilename(pos, "blk")))
                 break; // No block files left to reindex
             FILE *file = OpenBlockFile(pos, true);
             if (!file) break; // This error is logged in OpenBlockFile
+            uiInterface.InitMessage(_(tfm::format("Reindexing block from files... (blk%05u.dat)", (unsigned int)nFile).c_str()));
```

Comment:
Due to the internal logic of the `metrics thread` (responsible for the UI) this reindexing message is never shown because it gets continuously overwritten by the `printStats()` function.

I wouldn't remove the `InitMessage()` here at the moment, but it might be the case to investigate in the future how we can improve the update logic of the UI so that all the important information is shown. For my personal preference, I would like to see the normal block height counter increasing and the "reindex" message as well, so that the user can clearly understand that the blocks are not being downloaded from other peers.

Reply:
I don't agree with Paolo's analysis.
As shown in Jira ticket the message of reindexing was printed and stuck to the very same value till the reindexing finishes.
I believe several additions here introduced were not really necessary

Reply:
The message "Done loading" (which sets loaded == true in metrics thread) is not shown until chainActive.Tip() != NULL: and this condition happens only after reindex has completed (main thread stays in sleep wait every 10 milliseconds)

Reply:
Unfortunately `chainActive.Tip()` is NULL only during the processing of the headers, as soon as the first block is read from file and processed it's not NULL anymore, that's why we need such change.

Resolving the comment to proceed with the merge.